components/CostumerForm.js

```diff
@@ -0,0 +1,273 @@
+import { useState } from "react";
+import styled from "styled-components";
+
+export default function CostumerForm({
+  costumers,
+  handleAddCostumerForm,
+  handleAddCostumerFormSubmit,
+  deleteCostumer,
+  setCostumers,
+}) {
+  const [editingCostumer, setEditingCostumer] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    phoneNumber: "",
+    street: "",
+    zip: "",
+    city: "",
+  });
+
+  const [costumerEditId, setCostumerEditId] = useState(null);
+
+  const handleEditFormChange = (event) => {
+    event.preventDefault();
+    const fieldName = event.target.name;
+    const fieldValue = event.target.value;
+
+    const newFormData = { ...editingCostumer };
+    newFormData[fieldName] = fieldValue;
+    setEditingCostumer(newFormData);
+  };
+
+  const handleEditFormSubmit = (event) => {
+    event.preventDefault();
+
+    const editedCostumer = {
+      id: costumerEditId,
+      firstName: editingCostumer.firstName,
+      lastName: editingCostumer.lastName,
+      email: editingCostumer.email,
+      phoneNumber: editingCostumer.phoneNumber,
+      street: editingCostumer.street,
+      zip: editingCostumer.zip,
+      city: editingCostumer.city,
+    };
+
```

Comment:
It's best to transfer the edit function onto the card component. 
Also check if you are repeating yourself with handleEditClick, handleEditFormSubmit and handleEditFormChange, usually you won't need an onChange Event AND an onSubmit, its either or.
In your case onSubmit is enough :)